pkg/pillar/cmd/volumemgr/updatestatus.go

```diff
@@ -414,17 +414,16 @@ func doUpdateVol(ctx *volumemgrContext, status *types.VolumeStatus) (bool, bool)
 			status.Key(), status.DisplayName)
 		return false, true
 	}
-	changed := false
-	switch status.VolumeContentOriginType {
-	case zconfig.VolumeContentOriginType_VCOT_BLANK:
-		// XXX TBD
-		errStr := fmt.Sprintf("doUpdateVol(%s) name %s: Volume content origin type %v is not implemeted yet.",
+	if status.VolumeContentOriginType == zconfig.VolumeContentOriginType_VCOT_UNKNOWN {
+		// Unsupported volume content origin type
+		errStr := fmt.Sprintf("doUpdateVol(%s) name %s: Volume content origin type %v not supported",
 			status.Key(), status.DisplayName, status.VolumeContentOriginType)
 		status.SetErrorWithSource(errStr,
 			types.VolumeStatus{}, time.Now())
-		changed = true
-		return changed, false
-	case zconfig.VolumeContentOriginType_VCOT_DOWNLOAD:
+		return true, false
+	}
+	changed := false
+	if status.VolumeContentOriginType == zconfig.VolumeContentOriginType_VCOT_DOWNLOAD {
```

Comment:
Any particular reason the case statement on VolumeContentOriginType doesn't work with these changes? Couldn't you just add the UNKNOWN to the case statement or is there some other subtle change here?

Reply:
Yes, indeed, returned switch